R/shiny-shims.R

```diff
@@ -0,0 +1,47 @@
+shinyInstalled <- local({
+  detected <- FALSE
+  function() {
+    if (detected) {
+      return(TRUE)
+    }
+
+    if (nzchar(system.file(package = "shiny"))) {
```

Comment:
I don't know if this is your intent, but the result is only cached when shiny is installed; it will run this code every time when Shiny is not installed.

Reply:
Right, the idea is that if you install Shiny, you shouldn’t need to restart for crosstalk to start working. But if you uninstall Shiny while using crosstalk, I don’t have a lot of sympathy for you. 

(BTW I did some benchmarking and system.file is a lot slower when the package exists than when it doesn’t.)